### PR TITLE
add personal msg

### PR DESCRIPTION
**Pull request by Weii-Dev9**
## Overview
- Reference card [#10](https://trello.com/c/LT607LOY/10-朋友圈顶上有当前的用户的信息)

Why was this PR created / what was done?

- add personal message on the top of friends circle

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
1. [x] Makes your PR pass tests?
2. [x] PR attached to trello card
3. [x] Trello card moved into 'Review'

### CODE REVIEW

File: app/src/androidTest/java/com/thoughtworks/androidtrain/helloworld/ComposeTest.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @Test
+    fun profileTest() {
+        composeTestRule.setContent {
+            Image(
+                painter = painterResource(id = R.mipmap.background),
+                contentDescription = "背景图片",
```
Comment (yangyang-wen) [inline app/src/androidTest/java/com/thoughtworks/androidtrain/helloworld/ComposeTest.kt]: 看了一下，有多个背景图片，如果有多个是否可以抽出来，通过一个变量或是单独的一个文件来统一管理单元测试中的常量。

File: app/src/androidTest/java/com/thoughtworks/androidtrain/helloworld/ComposeTest.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                        Icon(
+                            painter = painterResource(id = R.mipmap.camera),
+                            contentDescription = "发表朋友圈"
+                        )
+                    }
+                }
+            )
+        }
+        composeTestRule.onNodeWithContentDescription("返回").assertIsDisplayed()
+        composeTestRule.onNodeWithContentDescription("返回").performClick()
```
Comment (yangyang-wen) [inline app/src/androidTest/java/com/thoughtworks/androidtrain/helloworld/ComposeTest.kt]: ’返回‘同理。

File: app/src/main/java/com/thoughtworks/androidtrain/helloworld/compose/TweetsScreen.kt [hunk 3]
```diff
@@ -160,6 +162,96 @@ private fun divider() {
...
+                    Icons.Default.ArrowBack,
+                    modifier = Modifier.size(actionIconSize)
+                        .padding(top = actionIconPaddingTop),
+                    contentDescription = stringResource(R.string.back)
+                )
+            }
+        },
+        actions = {
+            IconButton(onClick = { /*发表朋友圈*/ }) {
+                Icon(
```
Comment (yangyang-wen) [inline app/src/main/java/com/thoughtworks/androidtrain/helloworld/compose/TweetsScreen.kt:194]: 我理解，一般注释应当放在方法名的上面，放在里面个人感觉不合理

> /*发表朋友圈*/

File: app/src/main/java/com/thoughtworks/androidtrain/helloworld/compose/TweetsScreen.kt [hunk 4]
```diff
@@ -160,6 +162,96 @@ private fun divider() {
...
+
+@Composable
+private fun HeaderBackground() {
+
+    val backgroundHeight = dimensionResource(id = R.dimen.header_background_height)
+    val personalNicknameSize = dimensionResource(id = R.dimen.personal_nick_size)
+    val personalNicknamePaddingEnd = dimensionResource(id = R.dimen.personal_nick_padding_end)
+    val personalAvatarSize = dimensionResource(id = R.dimen.personal_avatar)
+    val personalAvatarPadding = dimensionResource(id = R.dimen.personal_avatar_padding)
+    val personalAvatarOffSetY = dimensionResource(id = R.dimen.personal_avatar_offset_y)
```
Comment (yangyang-wen) [inline app/src/main/java/com/thoughtworks/androidtrain/helloworld/compose/TweetsScreen.kt:213]: 是否可以简化一下，个人感觉变量过多. 将这个抽出来dimensionResource

File: app/build.gradle [hunk 5]
```diff
@@ -123,4 +123,7 @@ dependencies {
     implementation "com.google.dagger:hilt-android:2.44"
     kapt "com.google.dagger:hilt-android-compiler:2.44"
     implementation "androidx.hilt:hilt-navigation-compose:1.0.0"
+
+    androidTestImplementation("androidx.compose.ui:ui-test-junit4:1.3.3")
+    debugImplementation("androidx.compose.ui:ui-test-manifest:1.3.3")
```
Comment (yunlongwen) [inline app/build.gradle:128]: 为什么是debugImplementation？

File: app/src/main/java/com/thoughtworks/androidtrain/helloworld/compose/TweetsScreen.kt (hunk 3 above)
Comment (yangyang-wen) [reply app/src/main/java/com/thoughtworks/androidtrain/helloworld/compose/TweetsScreen.kt:194]: /*TODO:发表朋友圈*/

**Review by yunlongwen — APPROVED** [review]

**Pull request merged**